jerry-core/ecma/base/ecma-helpers.c

```diff
@@ -880,15 +940,22 @@ ecma_assert_object_contains_the_property (const ecma_object_t *object_p, /**< ec
                                           const ecma_property_t *prop_p) /**< ecma-property */
 {
 #ifndef JERRY_NDEBUG
-  ecma_property_t *prop_iter_p;
-  for (prop_iter_p = ecma_get_property_list (object_p);
-       prop_iter_p != NULL;
-       prop_iter_p = ECMA_GET_POINTER (ecma_property_t, prop_iter_p->next_property_p))
+  ecma_property_header_t *prop_iter_p = ecma_get_property_list (object_p);
+
+  while (prop_iter_p != NULL)
   {
-    if (prop_iter_p == prop_p)
+    JERRY_ASSERT (ECMA_PROPERTY_IS_PROPERTY_PAIR (prop_iter_p));
+
+    ecma_property_pair_t *prop_pair_p = (ecma_property_pair_t *) prop_iter_p;
+
+    if ((prop_pair_p->header.types + 0) == prop_p
+        || (prop_pair_p->header.types + 1) == prop_p)
```

Comment:
Please, implement this check with a loop. (Also, we are in debug code here, so performance is not an issue.)